app/logic/finder.rb

```diff
@@ -0,0 +1,36 @@
+class Finder
+
+  def initialize options
+    @options = Parser.new(options).perform
+  end
+
+  def perform
+    @query = find_all
+    @query = filter_by_author if options[:author]
+    @query = filter_by_phrase if options[:phrase].present?
+    @query = filter_by_status unless options[:status].nil?
+
+    @query
+  end
+
+  private
+
+  attr_reader :options, :query
+
+  def find_all
+    List.including_items.with_creator
+  end
+
+  def filter_by_author
+    query.where(users: {email: options[:author]})
+  end
+
+  def filter_by_phrase
+    query.joins(:items).where('items.text ILIKE ?', "%#{options[:phrase]}%")
+  end
```

Comment:
@sergeyradkevich, here is current sql:

      `SELECT "lists"."id" AS t0_r0, "lists"."title" AS t0_r1, "lists"."description" AS t0_r2, "lists"."created_at" AS t0_r3, "lists"."updated_at" AS t0_r4, "lists"."user_id" AS t0_r5,`
      `"items"."id" AS t1_r0, "items"."text" AS t1_r1, "items"."created_at" AS t1_r2, "items"."updated_at" AS t1_r3, "items"."list_id" AS t1_r4, "items"."done" AS t1_r5, "items"."position" AS t1_r6, "items"."completed_at" AS t1_r7,`
      `"users"."id" AS t2_r0, "users"."email" AS t2_r1, "users"."created_at" AS t2_r2, "users"."updated_at" AS t2_r3, "users"."encrypted_password" AS t2_r4, "users"."remember_created_at" AS t2_r5`
      `FROM "lists" INNER JOIN "items" ON "items"."list_id" = "lists"."id"`
      `LEFT OUTER JOIN "users" ON "users"."id" = "lists"."user_id"`
      `WHERE "users"."email" = $1 AND (items.text ILIKE '%click%') AND "items"."done" = 't'`              `ORDER BY "lists"."created_at" ASC  [["email", "vasyll@interlink-ua.com"]]`

I used `joins` combining with `includes`, because `includes` loads all the associations into the memory and I can use them in a template without new queries, and `joins` makes it possible to select fields from the associated (`items`) table. Otherwise (without `joins`) I get an error: 

      `PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "items"`